src/ll/parser.rs

```diff
@@ -559,6 +559,13 @@ impl Parser {
             TokenKind::String(_) => Ok(self.parse_string(token)),
             TokenKind::LongString(_) => self.parse_long_string(token),
             TokenKind::Identifier(_) => self.parse_identifier(token),
+            TokenKind::Underscore => {
+                Ok(self
+                  .ast
+                  .build_node(NodeKind::DiscardPattern, ())
+                  .with_location(token.location)
+                  .done())
+            },
```

Comment:
todo: Use the `parse_unit` function instead of building the node manually here:
```suggestion
            TokenKind::Underscore => Ok(self.parse_unit(token, NodeKind::DiscardPattern)),
```